lasagne/layers/recurrent.py

```diff
@@ -107,7 +107,11 @@ class CustomRecurrentLayer(MergeLayer):
         :class:`lasagne.layers.Layer` instance which connects input to the
         hidden state (:math:`f_i`).  This layer may be connected to a chain of
         layers, which must end in a :class:`lasagne.layers.InputLayer` with the
-        same input shape as `incoming`.
+        same input shape as incoming, except for the first dimension: When
+        ``precompute_input == True`` (the default), it must be
+        ``incoming.output_shape[0]*incoming.output_shape[1]`` or `None`; when
+        ``precompute_input == False``, it must be ``incoming.output_shape[0]``
+        or `None`.
```

Comment:
> Perhaps we need to do a docstring audit?

Yes, at some point we definitely need that. We also need to unify the parameter type conventions used throughout the docs. The downside is that this will touch a lot of code, so this should probably be done at a time where there are no open PRs.

> Which convention would you prefer I use in this PR?

I'd prefer the one with double backticks. Single backticks are for variables, double backticks are for code: https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#sections
But we can also leave that for a future audit, `recurrent.py` uses both single backticks and no backticks already. Unifying this is out of scope for this PR anyway.